### ISSUE TITLE
Barrier: Some actor does not collect barrier so that epoch can not commit

### ISSUE THREAD

**Issue by Little-Wallace** [singularity-data/risingwave#4270]
**Describe the bug**

Run ./scripts/launch_risedev_bench.sh  in https://github.com/singularity-data/tpch-bench and I found that compute-node all stop processing data in a few minutes. And the metrics `barrier-send-latency` does not show anything.

I add more log in compute-node and meta-node and I found that some actor does not run `LocalBarrierManager::collect` so that the rpc in `StreamService::barrier_complete` does not return result to meta-service and meta-service could not commit this epoch. It seems that some actor is blocked so that it can not do anything.

**To Reproduce**

deploy a new risingwave cluster in AWS EC2 as https://singularity-data.quip.com/EbTfA4xUdJ2z/Benchmark-RisingWave-Three-Easy-Steps (We need at least three compute-node) and run ./scripts/launch_risedev_bench.sh. Then you can observe you cluster are blocked.

**Expected behavior**

**Additional context**

**Comment by BugenZhao** [comment]
Which revision does this bug occur on?

**Comment by xxhZs** [comment]
> Which revision does this bug occur on?

The new Main

**Comment by xxhZs** [comment]
> Which revision does this bug occur on?

I found it in new Main branch yesterday

**Comment by BugenZhao** [comment]
> > Which revision does this bug occur on?
> 
> I found it in new Main branch yesterday

Do you have an exact commit hash?

**Comment by xxhZs** [comment]
> > > Which revision does this bug occur on?
> > 
> > 
> > I found it in new Main branch yesterday
> 
> Do you have an exact commit hash?

8ad51838db0f4bc013ad0f05a92421ba8f0200df

**Comment by BugenZhao** [comment]
I'm also encountering deadlock of "epoch can not commit" when running nexmark q5 with very large parallelism (not related to #4354). May be related to this.

**Comment by Little-Wallace** [comment]
@BugenZhao  I find that this bug may be caused by https://github.com/singularity-data/risingwave/pull/4045 with binary-search.

**Comment by BugenZhao** [comment]
> @BugenZhao  I find that this bug may be caused by https://github.com/singularity-data/risingwave/pull/4045 with binary-search.

Oops. Any further investigations?

**Comment by BugenZhao** [comment]
> @BugenZhao  I find that this bug may be caused by https://github.com/singularity-data/risingwave/pull/4045 with binary-search.

I've reviewed the implementation of remote input and found that creating client with moka might be suspicious, considering the bug we found in our LruCache last time. I've not checked how moka dedups inflight requests yet.

https://github.com/singularity-data/risingwave/blob/f5ab9b4e561ba2a2e8aaff062db9dc0fde44ecb4/src/stream/src/executor/exchange/input.rs#L125

A quick PoC might be creating the client outside of the stream and check whether there're still problems.
